### PR TITLE
Enhance classification task's model performance

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

1. Memory caching
    - Fix the bug that the memory cache handler tries to resize and cache an image although there is no space on its memory pool.
    - Fix the bug that `multiprocessing.Manager` can be shutdown not in the main thread (dataloader's worker thread).
    - Reuse the resized image during caching

2. Change image decoder from OpenCV to PIL
    - Change the default image color channel format as RGB and remove the BGR to RGB color channel format conversion in the data preprocessor.

3.  Set `pin_memory=True` in `DataLoader`.

4. Utilize a stacked tensor for a batch of images rather than a list of individual images.

   - This approach is necessary to eliminate the need for stacking image tensors in the main thread.

***
- **This is the performance comparison according to this PR**
image
***

### How to test
I updated our unit tests related to this change and the integration tests must guard the correct training pipeline.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — DISMISSED** [review]
LGTM

**Review by sungmanc — DISMISSED** [review]

**Comment by vinnamkim** [comment]
<del>Converted to draft to fix https://github.com/openvinotoolkit/training_extensions/actions/runs/7384134398/attempts/1?pr=2749</del>

File: src/otx/recipe/detection/atss_mobilenetv2.yaml [hunk 1]
```diff
@@ -115,7 +115,7 @@ model:
           - 255
           - 255
           - 255
-        bgr_to_rgb: true
+        bgr_to_rgb: false
```
Comment (jaegukhyun) [inline src/otx/recipe/detection/atss_mobilenetv2.yaml:118]: Should this value be always false or follow original mmX config?

File: src/otx/core/data/entity/action_classification.py [hunk 2]
```diff
@@ -72,6 +72,12 @@ def collate_fn(cls, entities: list[ActionClsDataEntity]) -> ActionClsBatchDataEn
             labels=[entity.labels for entity in entities],
         )
 
+    def pin_memory(self) -> ActionClsBatchDataEntity:
+        """Pin memory for member tensor variables."""
+        super().pin_memory()
+        self.labels = [label.pin_memory() for label in self.labels]
+        return self
+
```
Comment (jaegukhyun) [inline src/otx/core/data/entity/action_classification.py:80]: It seems ActionDetBatchDataEntity doesn't have this part

File: src/otx/recipe/detection/atss_mobilenetv2.yaml (hunk 1 above)
Comment (sungmanc) [reply src/otx/recipe/detection/atss_mobilenetv2.yaml:118]: After this PR, our data pipeline used the PIL.load(RGB) to load the image, however, mmX used the openCV(BGR). So we need to change the value according to the loading pipeline.

Comment (vinnamkim) [reply src/otx/recipe/detection/atss_mobilenetv2.yaml:118]: > After this PR, our data pipeline used the PIL.load(RGB) to load the image, however, mmX used the openCV(BGR). So we need to change the value according to the loading pipeline.

True

Comment (jaegukhyun) [reply src/otx/recipe/detection/atss_mobilenetv2.yaml:118]: So it is reverse of original mmX config right?

Comment (sungmanc) [reply src/otx/recipe/detection/atss_mobilenetv2.yaml:118]: Yes, if mmX will use the openCV loading.

File: src/otx/core/data/entity/action_classification.py (hunk 2 above)
Comment (vinnamkim) [reply src/otx/core/data/entity/action_classification.py:80]: Fixed in https://github.com/openvinotoolkit/training_extensions/pull/2749/commits/072f92dc6c8f36d67828a7cc5efe1ddbb74eacfe

File: src/otx/recipe/detection/atss_mobilenetv2.yaml (hunk 1 above)
Comment (vinnamkim) [reply src/otx/recipe/detection/atss_mobilenetv2.yaml:118]: > So it is reverse of original mmX config right?

Yes

**Review by jaegukhyun — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Pull request merged**